.github/workflows/presubmit.yml

```diff
@@ -0,0 +1,36 @@
+name: SmoothApp Presubmit Tests
+
+on: pull_request
+    
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@v1
+
+      # Setup flutter 
+      - name: "Setup Flutter"
+        uses: subosito/flutter-action@v1
+        with:
+          channel: 'stable' 
+          # flutter-version: '1.12.x'
+      
+      # Get dependencies.
+      - name: "Get dependencies"
+        run: ci/pub_upgrade.sh
```

Comment:
What is the difference between the scripts and the normal flutter commands and should we also run the scripts from the postsubmit so that it is more uniform?

Reply:
The main difference is that you can run them locally before pushing so you can correct issues before they get to the PR.

> should we also run the scripts from the postsubmit so that it is more uniform?

Good idea: I'll change things to also run them at appropriate points in the postsubmit script.